File: api/src/main/java/dev/obscuria/fragmentum/client/FragmentumClientRegistry.java
package dev.obscuria.fragmentum.client;

import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
import net.minecraft.world.inventory.tooltip.TooltipComponent;
import org.apache.commons.lang3.NotImplementedException;

import java.util.function.Function;

@SuppressWarnings("unused")
public interface FragmentumClientRegistry {

    static ClientRegistrar registrar(String modId) {
        throw new NotImplementedException();
    }

    static <T extends TooltipComponent> void registerTooltipComponent(Class<T> type, Function<T, ClientTooltipComponent> factory) {
        throw new NotImplementedException();
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/client/FragmentumClientRegistry.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.client

import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent
import net.minecraft.world.inventory.tooltip.TooltipComponent
import java.util.function.Function

object FragmentumClientRegistry {

    fun registrar(modId: String): ClientRegistrar = error("API stub")

    fun <T : TooltipComponent> registerTooltipComponent(
        type: Class<T>,
        factory: Function<T, ClientTooltipComponent>
    ): Unit = error("API stub")
}